### PR TITLE
CTL proxy for gopkg.in/alecthomas/kingpin.v2

### PR DESCRIPTION
**Pull request by dissoupov**

### CODE REVIEW

File: ctl/ctl.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ type Control interface {
 	Fail(msg string, err error) error
 
 	Parse(args []string) string
-	PopulateControl(c *kp.ParseContext) error
+	//PopulateControl(c *ParseContext) error
```
Comment (haykbaluyan) [inline ctl/ctl.go]: Should this be removed?

File: ctl/proxy.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import (
+	"io"
+
+	kp "gopkg.in/alecthomas/kingpin.v2"
+)
+
+// Action is a CTL action
+type Action func() error
+
+// ParseContext is a proxy interface to kingpin's *ParseContext
```
Comment (haykbaluyan) [inline ctl/proxy.go]: * is not needed in the comment.

File: ctl/proxy.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+// Application is a proxy interface to kingpin's *Application
+type Application interface {
+	Version(version string) Application
+	Parse(args []string) (command string, err error)
+	Command(name, help string) *CmdClause
+	Flag(name, help string) *FlagClause
+	Terminate(terminate func(int)) Application
+	ErrorWriter(w io.Writer) Application
+	UsageWriter(w io.Writer) Application
+	//	ParseContext(args []string) (*ParseContext, error)
```
Comment (haykbaluyan) [inline ctl/proxy.go]: should be removed?

File: ctl/proxy.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		Application: kp.New(name, help),
+	}
+}
+
+func (a *proxyapp) Terminate(terminate func(int)) Application {
+	a.Application.Terminate(terminate)
+	return a
+}
+
+func (a *proxyapp) Parse(args []string) (command string, err error) {
```
Comment (haykbaluyan) [inline ctl/proxy.go]: How is that exported method does not have a comment?

File: ctl/proxy.go [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+// NewApplication creates a new application instance
+func NewApplication(name, help string) Application {
+	return &proxyapp{
+		Application: kp.New(name, help),
+	}
+}
+
+func (a *proxyapp) Terminate(terminate func(int)) Application {
```
Comment (haykbaluyan) [inline ctl/proxy.go]: Same as below comment.

File: ctl/proxy.go [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+	return a
+}
+
+// Version adds a --version flag for displaying the application version.
+func (a *proxyapp) Version(version string) Application {
+	a.Application.Version(version)
+	return a
+}
+
+// ParseContext parses the given command line and returns the fully populated
```
Comment (haykbaluyan) [inline ctl/proxy.go]: Is this block needed?

File: ctl/proxy.go (hunk 4 above)
Comment (dissoupov) [reply ctl/proxy.go]: it's not `exported`. It's interface implementation on a private 
`(a *proxyapp)`

**Pull request merged**